File: frontend/cypress/test-solutions/group-anagrams.js
class Solution {
    groupAnagrams(strs) {
        const groups = new Map();
        
        for (const s of strs) {
            // Sort the string to create a key for anagrams
            const key = s.split('').sort().join('');
            
            if (!groups.has(key)) {
                groups.set(key, []);
            }
            groups.get(key).push(s);
        }
        
        return Array.from(groups.values());
    }
}

File: frontend/cypress/test-solutions/group-anagrams.java
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> groups = new HashMap<>();
        
        for (String s : strs) {
            // Sort the string to create a key for anagrams
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);
            
            groups.computeIfAbsent(key, k -> new ArrayList<>()).add(s);
        }
        
        return new ArrayList<>(groups.values());
    }
}